validation/start.go

```
@@ -0,0 +1,110 @@
+	if err != nil {
+		util.Fatal(err)
+	}
+	g := util.GetDefaultGenerator()
+	g.SetProcessArgs([]string{"sh", "-c", fmt.Sprintf("echo 'process called' >> %s", "/output")})
+	r.SetConfig(g)
+	output := filepath.Join(bundleDir, g.Spec().Root.Path, "output")
+
+	// start without id
+	err = r.Start()
+	wrap(t, err != nil, specerror.NewError(specerror.StartWithoutIDGenError, fmt.Errorf("start` operation MUST generate an error if it is not provided the container ID"), rspecs.Version), err)
+
+	// set id for the remaining tests
+	r.SetID(containerID)
+
+	// start a not `created` container - case one: non-exist container
+	err = r.Start()
+	wrap(t, err != nil, specerror.NewError(specerror.StartNotCreatedGenError, fmt.Errorf("attempting to `start` a container that is not `created` MUST generate an error"), rspecs.Version), err)
+
+	r.Create()
+	// start a `created` container
+	err = r.Start()
+	outputData, _ := ioutil.ReadFile(output)
+	// check the output
+	wrap(t, err == nil && string(outputData) == "process called\n", specerror.NewError(specerror.StartProcImplement, fmt.Errorf("`start` operation MUST run the user-specified program as specified by `process`"), rspecs.Version), err)
+
+	// start a not `created` container - case two: exist and `stopped`
+	util.WaitingForStatus(r, util.LifecycleStatusStopped, time.Second*10, time.Second*1)
+	err = r.Start()
+	// must generate an error
+	wrap(t, err != nil, specerror.NewError(specerror.StartNotCreatedGenError, fmt.Errorf("attempting to `start` a container that is not `created` MUST generate an error"), rspecs.Version), err)
+
+	outputData, _ = ioutil.ReadFile(output)
+	// must have no effect, it will not be something like 'process called\nprocess called\n'
+	wrap(t, string(outputData) == "process called\n", specerror.NewError(specerror.StartNotCreatedHaveNoEffect, fmt.Errorf("attempting to `start` a container that is not `created` MUST have no effect on the container"), rspecs.Version), err)
+
+	r.Delete()
```

You have another ignored error here.